### ISSUE TITLE
Google authenticated users should have their avatars uploaded

### ISSUE THREAD

**Issue by julianguyen** [ifmeorg/ifme#101]
After a user signs up using their Google account, they should have the option to upload their Google avatar as their if me avatar in the Account page.

**Comment by julianguyen** [comment]
Hey how is this going @aturkewi?

**Comment by aturkewi** [comment]
Hey @julianguyen ! It's going well. I know I started working on this a couple of weeks ago and I was hunting down the correct places to make the change. I haven't had a chance to finish it up, but I'm still looking at taking care of this issue. Just so I know, is there is a certain time you are looking to have this complete by? or simply the sooner the better?

**Comment by julianguyen** [comment]
Thanks for the update @aturkewi! Within October should be fine :D Let me us know on Slack if you need any help!

**Comment by julianguyen** [comment]
Any updates @aturkewi?

**Comment by alexiamcdonald** [comment]
This has been up for a while. I can take a look at it.

Were any code branches for this issue previously?

**Comment by baohouse** [comment]
@alexiamcdonald Your help is much appreciated. I don't think there's a branch anymore; it was seen deleted in #271 nearly a year ago.

**Comment by alexiamcdonald** [comment]
@baohouse The branch was just for a readme info which would be out of date now. There is a bunch of Google related issues

I can see @abrophy is already working on one issue [#586 Calendar Reminders not working](https://github.com/julianguyen/ifme/issues/586).

@abrophy anything else I need to be aware of? Do you think we should bundle up the Google issues together?

**Comment by abrophy** [comment]
I'm reading through the documentation for the new ruby google-client. Grouping anything that relies on it sounds like a good idea.

**Comment by justinwalker4179** [comment]
Is this issue open to be worked on still, or did it get grouped in with another google-client issue?

**Comment by julianguyen** [comment]
Hey @justinwalker4179, this issue is still open! :)

**Comment by janessatran** [comment]
Hi @julianguyen 

@bracken and I were wondering how we should handle the Google Avatar images on the user profiles. Do you want us to upload the image to Cloudinary like it does when a user uploads a profile picture manually, or should we have a separate "third party login profile image url" and the user choose which image they want to show? Hopefully that makes sense, but let me know if you're confused by what we're asking!

**Comment by julianguyen** [comment]
Good questions @janessatran, yes we should upload these images to Cloudinary! So I think we should automatically use the Google avatar image if it's available.

**Comment by janessatran** [comment]
Thanks for the response @julianguyen! We were wondering which route to take for handling the Google avatar image and outlined a few options below.
1. Always set "If me" avatar to Google avatar image if available.
2. Only set "If me" avatar to Google avatar image the first time they login via Google.
3. Only set "If me" avatar to Google avatar image if the "If me" avatar is blank.
4. Ask user after login if they want to change their "If me" avatar to the Google avatar.
5. Remember the Google avatar image and give user option to use it on the profile page.
6. Any other options you prefer

What are your thoughts?

**Comment by julianguyen** [comment]
Hey @janessatran, thanks for writing out these options! So I think we want to do the combination of options 2 and 3.

Set the avatar to the Google avatar after someone logs in via Google Auth if the avatar is blank.

**Comment by janessatran** [comment]
Hi @julianguyen, thanks for the feedback! We (@bracken and I) went ahead and implemented the feature with a combination of options 2 and 3 and made a PR. We wanted to bring up that if a user does not want a profile image at all, this will force them to always have their Google avatar image uploaded. Do you want us to handle that somehow? If so, how? :)

**Comment by julianguyen** [comment]
Thanks for the awesome work! Great questions. I think we want to give the user the option to remove the uploaded Google avatar image and opt for the default placeholder image.

**Comment by janessatran** [comment]
Hey @julianguyen, this is what we're thinking of doing, what do you think?

![image](https://user-images.githubusercontent.com/17259147/62092364-80c8c780-b23a-11e9-8eeb-a1f4e75527e9.png)

**Comment by julianguyen** [comment]
So sorry for the late response @janessatran, looks good to me!

I think I would rephrase "Don't use third party avatar" to "Don't use my Google avatar".

**Comment by bracken** [comment]
@julianguyen OK, we'll do that. Thanks! 
We used "third party avatar" since we saw tickets for adding logins via some other providers like facebook/twitter and thought this could be generic. The back-end implementation is also generic so that those other logins can easily take advantage of it as well.

**Comment by julianguyen** [comment]
Ooh that's a good point! Hmm ok, how about we rephrase it to, "Don't use a third-party avatar" then?

**Comment by janessatran** [comment]
@julianguyen no need to apologize :) Thank you for the feedback!
